### PR TITLE
Add config option to add caller="file:line" field

### PR DESCRIPTION
**Pull request by elv-todd**
adds caller info

### CODE REVIEW

**Review by elv-gilles — COMMENTED** [review]
I would suggest to rather add a `caller` field to the config, similar to `GoRoutineID` (but `bool` rather than `*bool`).

See:  [log_config](https://github.com/eluv-io/log-go/blob/9d221d52c2e784fe1e5e06fcf9c0de54b71a3ca0/log_config.go#L23) and where it is used (in functions [fields](https://github.com/eluv-io/log-go/blob/9d221d52c2e784fe1e5e06fcf9c0de54b71a3ca0/log.go#L522-L527) and [mergeConfig](https://github.com/eluv-io/log-go/blob/9d221d52c2e784fe1e5e06fcf9c0de54b71a3ca0/log.go#L251))

**Comment by elv-todd** [comment]
> I would suggest to rather add a `caller` field to the config, similar to `GoRoutineID` (but `bool` rather than `*bool`).

sure, updated.  good suggestion.

GoRoutineID is a *bool for mergeConfig.  I think I have to use a *bool to support that, or just assume true overrides false.

File: log_config.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ type Config struct {
 	// Include go routine ID as 'gid' in logged fields
 	GoRoutineID *bool `json:"go_routine_id,omitempty"`
 
+	// Include caller info (file:line) as 'caller' in logged fields
+	Caller bool `json:"caller,omitempty"`
```
Comment (elv-todd) [inline log_config.go]: seems weird we don't call these IncludeGoRoutineId / WithGoRoutineId or whatever, they're directives not things.

**Comment by elv-gilles** [comment]
> GoRoutineID is a *bool for mergeConfig. I think I have to use a *bool to support that.

Yes, you're right (my bad)

**Review by elv-gilles — APPROVED** [review]

**Pull request merged**